autogpt_platform/backend/schema.prisma

```diff
@@ -110,6 +112,57 @@ model AgentPreset {
   @@index([userId])
 }
 
+enum NotificationType {
+  AGENT_RUN
+  ZERO_BALANCE
+  LOW_BALANCE
+  BLOCK_EXECUTION_FAILED
+  CONTINUOUS_AGENT_ERROR
+  DAILY_SUMMARY
+  WEEKLY_SUMMARY
+  MONTHLY_SUMMARY
+}
+
+model NotificationEvent {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @default(now()) @updatedAt
+
+  UserNotificationBatch   UserNotificationBatch? @relation(fields: [userNotificationBatchId], references: [id])
+  userNotificationBatchId String?
+
+  type NotificationType
+  data Json
```

Comment:
Are we expecting the data field to be dynamic with an unknown set of keys used?

Reply:
Yes, similar pattern used throughout the system

Reply:
Can you elaborate for me on how the data is expected to be dynamic. What do we believe will be stored here?

Reply:
I'd really like to understand this and why it's happening in the code base. 

Can we open a discussion about it on discord?